2015/18/grid.js

```diff
@@ -46,34 +31,30 @@ class Grid {
             for (let y = 0; y < this.grid.length; y++) {
                 for (let x = 0; x < this.grid[0].length; x++) {
                     let cell = this.grid[y][x];
-                    let neighbors = this.getNeighbors(x, y);
 
+                    let neighbors = this.getNeighbors(x, y);
                     let neighbors_on = 0;
                     let neighbors_off = 0;
 
                     neighbors.forEach(n => {
-                        if (n.on) neighbors_on++;
+                        if (n) neighbors_on++;
                         else neighbors_off++;
                     });
 
-                    if (cell.on) {
+                    if (cell) {
                         // A light which is _on_ stays on when 2 or 3 neighbors are on,
                         // and turns off otherwise.
                         new_grid_state[y][x] = neighbors_on === 2 || neighbors_on === 3;
                     } else {
                         // A light which is _off_ turns on if exactly 3 neighbors are on,
                         // and stays off otherwise.
-                        new_grid_state[y][x] = neighbors_off === 3;
+                        new_grid_state[y][x] = neighbors_on === 3;
```

Comment:
This was the actual bugfix. I didn't need to remove the `Lights` class but whatever, it made things a bit simpler, and I bet a tiny bit faster for the program too.